osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/DlmsConnector.java

```diff
@@ -20,6 +39,27 @@
 public abstract class DlmsConnector {
   private static final Logger LOGGER = LoggerFactory.getLogger(DlmsConnector.class);
 
+  private static final Map<String, FunctionalExceptionType> errorMap = new HashMap<>();
+
+  static {
+    errorMap.put("AUTHENTICATION_ERROR", AUTHENTICATION_ERROR);
+    errorMap.put("AUTHENTICATION_REQUIRED", AUTHENTICATION_REQUIRED);
+    errorMap.put("Connection refused", CONNECTION_REFUSED);
+    errorMap.put("Connection reset", CONNECTION_RESET);
+    errorMap.put("Connection timed out", CONNECTION_TIMED_OUT);
+    errorMap.put("CONNECTION_ESTABLISH_ERROR", CONNECTION_ESTABLISH_ERROR);
+    errorMap.put("Socket was closed by remote host.", SOCKET_CLOSED_BY_REMOTE);
+    errorMap.put("Unable to decypher/decrypt xDLMS pdu", UNABLE_TO_DECYPHER);
+    errorMap.put("WRAPPER_HEADER_INVALID_VERSION", WRAPPER_HEADER_ERROR);
+    errorMap.put("WRAPPER_HEADER_INVALID_SRC_DEST_ADDR", WRAPPER_HEADER_ERROR);
+    errorMap.put("WRAPPER_HEADER_INVALID_PAYLOAD_LENGTH", WRAPPER_HEADER_ERROR);
+    errorMap.put("WRAPPER_HEADER_INVALID", WRAPPER_HEADER_ERROR);
```

Comment:
klopt dit dat het allemaal WRAPPER_HEADER_ERROR zijn? Gaat hier dan geen info verloren?

Reply:
if so all error messages contain "WRAPPER_HEADER_INVALID" and others are therefor redundant

Reply:
That's true. Removed all map entries, except for the last one. No info will be lost, the original error message will be copied.